storage_service/locations/models/archipelago.py

```diff
@@ -0,0 +1,259 @@
+        try:
+            with open(source_path, "rb") as file:
+                response = requests.post(
+                    url,
+                    data=file,
+                    headers=headers,
+                    auth=(self.archipelago_user, self.archipelago_password),
+                )
+                response.raise_for_status()
+
+            if response.status_code == 201:
+                LOGGER.info("AIP file uploaded successfully!")
+                response_json = response.json()
+                fid = (
+                    response_json.get("data", {})
+                    .get("attributes", {})
+                    .get("drupal_internal__fid")
+                )
+                return fid
+            else:
+                LOGGER.error(
+                    f"File upload failed with status code {response.status_code}: {response.text}"
+                )
+        except (OSError, requests.exceptions.RequestException) as e:
+            LOGGER.error("Error during AIP upload to archipelago %s", str(e))
+
+    def extract_title_from_mets_xml(self, xml_string):
+        """Retrieves title from METs file or creates title from file name"""
+        try:
+            root = etree.fromstring(xml_string)
+            namespaces = utils.NSMAP
+            title_element = root.find(
+                ".//mets:dmdSec/mets:mdWrap/mets:xmlData/dcterms:dublincore/dc:title",
+                namespaces=namespaces,
+            )
+            if title_element is not None:
+                return title_element.text.strip()
+        except Exception as e:
+            LOGGER.error("Error extracting title from METS XML: %s", str(e))
+        return None
+
+    def get_dc_metadata(self, xml_string):
+        """Extracts Dublin Core metadata from METS file"""
+        try:
+            root = etree.fromstring(xml_string)
+            namespaces = utils.NSMAP
+            dc_fields = {}
+            for dc_element in root.findall(".//dc:*", namespaces=namespaces):
+                field_name = dc_element.tag.split("}")[-1]
+                if field_name == "title":
+                    continue
+                appended_field = "field_" + field_name
+                field_value = dc_element.text.strip() if dc_element.text else None
+                LOGGER.info(
+                    f"dc value added which is {field_value} where the field is {appended_field}"
+                )
+                dc_fields[appended_field] = field_value
+            strawberry = json.dumps(dc_fields)
+            LOGGER.info(f"strawberry json is {strawberry}")
+            return strawberry
+
+        except Exception as e:
+            LOGGER.error("Error extracting dc fields %s", str(e))
+        return None
+
+    @staticmethod
+    def _get_mets_el(package_type, output_dir, input_path, dirname, aip_uuid):
+        """Locate, extract (if necessary), XML-parse and return the METS file
+        for this package.
+        """
+        if package_type == "AIP":
```

Comment:
```suggestion
        
```
We are focusing on AIP right now. So, We don't think `package_type` attribute is in need.